includes/Modules/Analytics.php

```diff
@@ -1102,15 +1054,6 @@ function( $property ) use ( $urls ) {
 					}
 
 					$result = array_merge( $response, $properties );
```

Comment:
The current logic is (and FWIW has been for a while) a bit problematic: We only include `properties` in the response for the account that we've determined to be the "correct account". Since we now no longer perform the existing tag detection here (which is good!), the account found here might actually not be the right one (the existing tag detected in JavaScript might be from a different account). In this case the `properties` would still contain properties from the other account.

Would there be any downside to including properties for all accounts? We query them already anyway in https://github.com/google/site-kit-wp/pull/311/files#diff-82cb88ae0faf21fb5e86f13d2458e589R1028, so no request overhead here. It would make much more sense also from a data structure perspective. Maybe nest each account's `properties` as an item within that account's data in `accounts`? @lucymtc What do you think? We could create a separate issue and prioritize it.

Reply:
@felixarntz we get properties data foreach account until account id is found by matching URL https://github.com/google/site-kit-wp/blob/8f9f02d0e9e9f9da3cb43edb98a2957a05c37353/includes/Modules/Analytics.php#L1041  your suggestion of having properties nested for each account makes sense but I don't know how many accounts a user could have and how long it would take to fetch all properties for each account regardless of the account being found. Do you think this wouldn't be a problem if user has large number of accounts?

Reply:
@lucymtc That's a fair question. Generally, I'd argue this would be a problem already today - if the matching account is the last account, we'd run into the same issue.

I opened #341 to address this shortcoming. It's related, but I think with the changes I suggest in the overall review, we can get this ready in advance of that separate problem. It is crucial though that we can assure the properties returned by `get-accounts` actually apply to the existing tag if there is one, which right now is not the case.